src/graph.js

```diff
@@ -0,0 +1,247 @@
+		this.canvas = config.canvas;
+
+		this.minX = config.minX || 0;
+		this.minY = config.minY || 0; // for use if tickY isn't set
+
+		this.maxX = config.maxX || 10;
+		this.maxY = config.maxY || 10; // for use if tickY isn't set
+
+		// Space on left until X's 0 starts (axis and label are drawn in this space)
+		this.marginX = config.maginX || 60;
+		// Space on bottom until Y's 0 starts (axis and label are drawn in this space)
+		this.marginY = config.marginY || 45;
+		// Space on top of graph, usually just makes sure things don't get cut off
+		this.marginTop = config.marginTop || 20;
+		// Space to the right of the graph, just prevents tings from cutting off
+		this.marginRight = config.marginRight || 15;
+
+		this.labelX = config.labelX || "x";
+		this.labelY = config.labelY || "f(x)";
+
+		this.color = config.color || "green";
+
+		this.equation = config.equation;
+		// How many samples to take inbetween each point
+		this.resolution = config.resolution || 100;
+
+		this.tickX = config.tickX || 1;
+		this.iteration = (this.maxX - this.minX) / this.resolution;
+		// If tickY is set use that, otherwise we recalculate tickY, minY, and maxY based on equation and min/max X
+		if(config.tickY)
+		{
+			this.tickY = config.tickY;
+		}
+		else {
+			this.calcY();
+		}
+
+		// constants
+		this.axisColor = '#aaa';
+		this.font = '12pt Calibri';
+		this.tickSize = 5;
+
+		// relationships
+		this.context = this.canvas.getContext('2d');
+
+		this.rangeX = this.maxX - this.minX;
+		this.rangeY = this.maxY - this.minY;
+
+		// Scale used to map calcuated X,Y -> Canvas X,Y for graphing
+		this.scaleX = (this.canvas.width - this.marginX - this.marginRight) / this.rangeX;
+		this.scaleY = (this.canvas.height - this.marginY - this.marginTop) / this.rangeY;
+
+		this.context.clearRect(0, 0, this.canvas.width, this.canvas.height);
+		// draw x and y axis
+		this.drawAxes();
+	}
+
+	// Figure out ticks on Y axis
+	calcY() {
+		let max = Number.NEGATIVE_INFINITY;
+		let min = Number.POSITIVE_INFINITY;
+		for(let x = this.minX; x < this.maxX; x += this.tickX) {
+			let y = this.equation(x);
+			if(y < min) min = y;
+			if(y > max) max = y;
+		}
+		let range = max - min;
+		let tickRange = range / 5.0;
+		let pow10x = Math.pow(10, Math.ceil(Math.log(tickRange) - 1));
+		let roundedTickRange = Math.ceil(tickRange / pow10x) * pow10x;
+
+		this.minY = Math.floor(min * Math.pow(10, 2)) / Math.pow(10, 2);
+		this.maxY = Math.ceil(max * Math.pow(10, 2)) / Math.pow(10, 2);
+		this.tickY = Number(roundedTickRange.toFixed(2));
+	}
+
+	// Draw axes and labels for them
+	drawAxes(){
+		let context = this.context;
+		context.resetTransform();
+		this.transformContext();
+		this.drawXAxis();
+		this.drawYAxis();
+		this.drawText(this.labelX, this.getPixelX((this.maxX + this.minX)/2), this.marginY / 3);
+		// draw this one rotated
+		context.save();
+		context.resetTransform();
+		context.translate(5, this.canvas.height - this.getPixelY((this.maxY + this.minY) / 2));
```

Comment:
```suggestion
		context.translate(10, this.canvas.height - this.getPixelY((this.maxY + this.minY) / 2));
```

Unrelated to this PR really, but noticed the f(x) was being chopped off, and this fixes it

Reply:
Otherwise LGTM